velox/exec/TopN.cpp

```diff
@@ -59,12 +75,28 @@ void TopN::addInput(RowVectorPtr input) {
       newRow = data_->initializeRow(topRow, true /* reuse */);
     }
 
-    for (auto col = 0; col < input->childrenSize(); ++col) {
+    data_->initializeFields(newRow);
+    for (const auto col : sortingKeyColumns_) {
       data_->store(decodedVectors_[col], row, newRow, col);
     }
 
+    passedRows[newRow] = row;
     topRows_.push(newRow);
   }
+
+  if (!passedRows.empty() && !nonKeyColumns_.empty()) {
+    SelectivityVector needsDecode(input->size(), false);
```

Comment:
This loop is slow and is missing updateBound() I suggest to always decode for all rows by not passing SelectivityVector at all.

```
  /**
   * Set whether given index is selected. updateBounds() need to be called
   * explicitly after setValid() call, it can be called only once after multiple
   * setValid() calls in a row.
   */
  void setValid(vector_size_t idx, bool valid) {
    VELOX_DCHECK_LT(idx, bits_.size() * sizeof(bits_[0]) * 8);
    bits::setBit(bits_.data(), idx, valid);
    allSelected_.reset();
  }
```

Reply:
Removed this and now decode all rows, thanks!